p2p/server.go

```diff
@@ -876,6 +885,22 @@ func (srv *Server) listenLoop() {
 	}
 }
 
+func (srv *Server) checkInboundConn(fd net.Conn, remoteIP net.IP) error {
+	if remoteIP != nil {
+		// Reject connections that do not match NetRestrict.
+		if srv.NetRestrict != nil && !srv.NetRestrict.Contains(remoteIP) {
+			return fmt.Errorf("not whitelisted in NetRestrict")
+		}
+		// Reject Internet peers that try too often.
+		srv.inboundHistory.expire(time.Now())
+		if !netutil.IsLAN(remoteIP) && srv.inboundHistory.contains(remoteIP.String()) {
+			return fmt.Errorf("too many attempts")
+		}
+		srv.inboundHistory.add(remoteIP.String(), time.Now().Add(inboundThrottleTime))
```

Comment:
Is this a good idea to add at this point? If I understand correctly, this is happening immediately after an inbound TCP connection? If so, it might not be hard for an attacker to continuously send TCP requests from spoofed IPs of Ethereum peers to eclipse this node from the network. Might it be safer to add the entry at some point later, after the handshake?